### PR TITLE
Update react native docs for the removed APIs

### PR DESCRIPTION
**Pull request by yangyansong-adbe**

### CODE REVIEW

File: foundation-extensions/mobile-core/lifecycle/README.md [hunk 1]
```diff
@@ -446,21 +446,10 @@ using Com.Adobe.Marketing.Mobile;
 {% tab title="React Native" %}
 **Registering the extension with Core:**
 
-When using React Native, registering Lifecycle with Mobile Core should be done in native code which is shown under the Android and iOS tabs.
+When using React Native, registering Lifecycle with Mobile Core should be done in native code which is shown under the Android and iOS(ACP 2.x) tabs.
 
-> Note: Implementing Lifecycle via JavaScript may lead to inaccurate Lifecycle metrics, therefore we recommend implementing Lifecycle in native Android and iOS code. However, these APIs are still provided in JavaScript to support flexible Lifecycle implementations.
+> Note: Implementing Lifecycle via JavaScript may lead to inaccurate Lifecycle metrics, therefore we recommend implementing Lifecycle in native Android and iOS code.
```
Comment (cacheung) [inline foundation-extensions/mobile-core/lifecycle/README.md:451]: Should we remove this Note section?

File: foundation-extensions/mobile-core/lifecycle/lifecycle-api-reference.md [hunk 2]
```diff
@@ -186,13 +186,7 @@ ACPCore.lifecycleStart(["state": "appResume"])
...
 #### JavaScript
 
-Note: Implementing Lifecycle via JavaScript may lead to inaccurate Lifecycle metrics, therefore we recommend implementing Lifecycle in native Android and iOS code. However, these APIs are still provided in JavaScript to support flexible Lifecycle implementations.
-
-**Syntax**
-
-```jsx
-lifecycleStart(additionalContextData?: { string: string });
-```
+Note: Implementing Lifecycle via JavaScript may lead to inaccurate Lifecycle metrics, therefore we recommend implementing Lifecycle in native Android and iOS code.
```
Comment (cacheung) [inline foundation-extensions/mobile-core/lifecycle/lifecycle-api-reference.md:189]: Should we change to
When using React Native, registering Lifecycle with Mobile Core should be done in native code which is shown under the Android and iOS(ACP 2.x) tabs.

File: foundation-extensions/mobile-core/lifecycle/lifecycle-api-reference.md [hunk 3]
```diff
@@ -333,13 +327,7 @@ ACPCore.lifecyclePause()
...
 #### JavaScript
 
-> Note: Implementing Lifecycle via JavaScript may lead to inaccurate Lifecycle metrics, therefore we recommend implementing Lifecycle in native Android and iOS code. However, these APIs are still provided in JavaScript to support flexible Lifecycle implementations.
-
-**Syntax**
-
-```jsx
-lifecyclePause();
-```
+> Note: Implementing Lifecycle via JavaScript may lead to inaccurate Lifecycle metrics, therefore we recommend implementing Lifecycle in native Android and iOS code.
```
Comment (cacheung) [inline foundation-extensions/mobile-core/lifecycle/lifecycle-api-reference.md:330]: Should we change to 
When using React Native, registering Lifecycle with Mobile Core should be done in native code which is shown under the Android and iOS(ACP 2.x) tabs.

**Review by cacheung — APPROVED** [review]

**Comment by emdobrin** [comment]
@yangyansong-adbe left a small comment, this one seems good to after you fix that

File: foundation-extensions/mobile-core/lifecycle/README.md [hunk 4]
```diff
@@ -446,21 +446,8 @@ using Com.Adobe.Marketing.Mobile;
 {% tab title="React Native" %}
 **Registering the extension with Core:**
 
-When using React Native, registering Lifecycle with Mobile Core should be done in native code which is shown under the Android and iOS tabs.
+When using React Native, registering Lifecycle with Mobile Core should be done in native code which is shown under the Android and iOS(ACP 2.x) tabs.
```
[the remark this answers is not in the document]
Comment (yangyansong-adbe) [reply foundation-extensions/mobile-core/lifecycle/README.md]: @emdobrin  I have addressed the above comment in the lifecycle's README and API-REFERENCE pages.

**Review by emdobrin — APPROVED** [review]

**Pull request merged**